java/kafka/deployment-all.yaml

```diff
@@ -0,0 +1,154 @@
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+      strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  partitions: 12
+---
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaTopic
+metadata:
+  name: my-topic-reversed
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  partitions: 12
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: java-kafka-producer
+  name: java-kafka-producer
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: java-kafka-producer
+  template:
+    metadata:
+      labels:
+        app: java-kafka-producer
+    spec:
+      containers:
+      - name: java-kafka-producer
+        image: docker.io/owenerz/java-kafka-producer
+        env:
+         # - name: OPENTRACING_ENABLED
+         #   value: false
+          - name: BOOTSTRAP_SERVERS
+            value: my-cluster-kafka-bootstrap:9092
+          - name: TOPIC
+            value: my-topic
+          - name: DELAY_MS
+            value: "1000"
+          - name: LOG_LEVEL
+            value: "INFO"
+          - name: MESSAGE_COUNT
+            value: "1000000"
+          - name: JAEGER_SERVICE_NAME
+            value: kafka
+          - name: OTEL_SERVICE_NAME
+            value: kafka-otel
+          - name: OTEL_TRACES_EXPORTER
+            value: jaeger
+          - name: OTEL_METRICS_EXPORTER
+            value: none
+          - name: OTEL_EXPORTER_JAEGER_ENDPOINT
+            value: http://my-jaeger-collector:14250
+          - name: OTEL_LOG_LEVEL
+            value: debug
```

Comment:
this is mixing env vars used for OpenTracing (see `JAEGER_SERVICE_NAME`) and the others used by OpenTelemetry. They cannot live together.
I am also not sure we want the default deployment with all these env var set. It makes everything more complex.
We would need more an example of `deployment-open-tracing` and `deployment-open-telemetry` to show how using tracing.